test/e2e/deployment.go

```diff
@@ -56,9 +56,17 @@ const (
 
 var _ = framework.KubeDescribe("Deployment", func() {
 	f := framework.NewDefaultFramework("deployment")
```

Comment:
@kargakis You can move the "AfterEach" block  before you create the framework, this will let you register the cleanup func before the framework register. I just try it, and works well. If you don't have time update this pr, I can help update it.